JP-3345: Implement mingrad method for MIRI MRS pixel replacement

drlaw1558 opened the pull request:
Resolves [JP-3345](https://jira.stsci.edu/browse/JP-3345) by implementing a local pixel replacement alternative for the MIRI MRS.

Closes #7822 

**Checklist for maintainers**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [ ] updated or added relevant tests
- [x] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

tapastro reviewed (approved):
Looks good! Thanks for contributing an upgrade to the step 🙂  I will leave merging to the bosses.

hbushouse commented on jwst/pixel_replace/pixel_replace.py:
```diff
@@ -326,6 +333,104 @@ def fit_profile(self, model):
+        -------
+        model_replaced : DataModel
+            DataModel with flagged bad pixels now flagged with
+            TO-BE-DETERMINED and holding a flux value estimated
```
Don't we now have a DQ name that's been appropriated for this? If so, the comment could be updated.

drlaw1558 replied:
Yes, FLUX_ESTIMATED.  I'll make this change here and in the original code for the default method.

hbushouse commented on jwst/pixel_replace/pixel_replace.py:
```diff
@@ -326,6 +333,104 @@ def fit_profile(self, model):
+            except Exception:
+                pass
+
+        log.info(f"Number of pixels replaced: {nreplaced}")
```
I recall seeing similar log statements in the upper-level routine. Are we getting double reporting (or at least more than necessary)?

hbushouse commented:
There's one minor code style error that needs fixing:
jwst/pixel_replace/pixel_replace.py:392:49: E712 [*] Comparison to `False` should be `cond is False` or `if not cond:`

hbushouse commented:
Regression test run started at https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/875

drlaw1558 commented:
@hbushouse I wasn't sure entirely what to do about that style error on line 392.  When I tried replacing == with 'is' it busted the entire routine and it never flagged anything, returning a zero-length set from the np.where.  'is' doesn't seem to be an appropriate replacement for == in this case.  Is there some different syntax that should be used?

hbushouse commented:
> @hbushouse I wasn't sure entirely what to do about that style error on line 392. When I tried replacing == with 'is' it busted the entire routine and it never flagged anything, returning a zero-length set from the np.where. 'is' doesn't seem to be an appropriate replacement for == in this case. Is there some different syntax that should be used?

In that case we'll "call shenanigans" (to quote J. Eisenhamer) on the code style routine and override it to accept the as is code.

drlaw1558 commented:
Ok, cleaned up some outdated doc stuff and moved the higher-level logging statement out of the else clause so that it applies regardless of which method is used instead of having dedicated log statements in the subroutine.

hbushouse reviewed (approved):
LGTM
